community/values/src/main/java/org/neo4j/values/storable/DurationValue.java

```diff
@@ -226,6 +227,41 @@ else if ( seconds > 0 && nanos < 0 )
         this.nanos = (int) nanos;
     }
 
+    public int compareTo( DurationValue other )
+    {
+        return Comparator.comparingLong(DurationValue::averageLengthInNanos)
+                .thenComparingLong( DurationValue::getMonths )
+                .thenComparingLong( DurationValue::getDays )
+                .thenComparingLong( DurationValue::getSeconds )
+                .thenComparingInt( DurationValue::getNanos ).compare( this, other );
+    }
+
+    private long averageLengthInNanos()
+    {
+        return this.nanos + this.seconds * NANOS_PER_SECOND + this.days * NANOS_PER_SECOND * SECONDS_PER_DAY +
+                this.months * NANOS_PER_SECOND * (long) (SECONDS_PER_DAY * AVERAGE_DAYS_PER_MONTH);
```

Comment:
This will overflow for durations longer than 292 years, which isn't very long in the grand scheme of things.